### PR TITLE
Add configuration for ExecutionContext

### PR DESCRIPTION
**Pull request by phaller**
This adds configuration based on JVM system properties to ExecutionContexts. Patch submitted by @viktorklang 

Review by @axel22 and @heathermiller

### CODE REVIEW

**Comment by axel22** [comment]
lgtm

File: src/library/scala/concurrent/impl/ExecutionContextImpl.scala [hunk 1]
```diff
@@ -43,17 +43,31 @@ private[scala] class ExecutionContextImpl private[impl] (es: Executor, reporter:
...
-  def createExecutorService: ExecutorService = try { new ForkJoinPool(
-          Runtime.getRuntime.availableProcessors(), //FIXME from config
-          forkJoinPoolThreadFactory,
-          null, //FIXME we should have an UncaughtExceptionHandler, see what Akka does
-          true) //FIXME I really think this should be async...
+
+  def createExecutorService: ExecutorService = try {
+    def getInt(name: String, f: String => Int): Int =
+      try f(System.getProperty(name)) catch { case e: Exception => Runtime.getRuntime.availableProcessors }
+    def range(floor: Int, desired: Int, ceiling: Int): Int =
```
Comment (jsuereth) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:50]: The default for any Int is available processors?

I have alternative property passing code that looks more like:

``` scala

def parseInt(x: String) = 
  for {
    value <- Option(x) 
    number <- util.control.Exception.catching(classOf[NumberFormatException) opt value.toInt
  } yield number
def getInt(name: String) = parseInt(System.getProperty(name)) orElse parseInt(System.getenv(name))
def getIntOr(name: String, default: => Int) = getInt(name) getOrElse default
```

You can, of course, clean that up a bit.  The core idea though is to also allow environment variables to be used.  It gives a little bet extra flexibility.

In any case, food for future thought.  I'd hate for us to string a bunch of code like this through the standard library if we can consolidate a bit more.

I'd be nice if we had a standard `SystemProperties` object somewhere where you could call things like getInt, getFloat, getString etc.

Comment (axel22) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:50]: We have a `SystemProperties` in the `sys` package, right? We could extend the functionality there.

Comment (jsuereth) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:50]: Yep, perfect place for it.  I forgot where to find it and made a dumb assumption.

Looks like `SystemProperties` needs an `int` method to go along with `bool` and `str`.

Comment (viktorklang) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:50]: I'd vote for a typeclass approach.

Comment (phaller) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:50]: Feel free to submit a new pull request with the proposed changes if you want it to go into the milestone. I'm off the grid for a few days, so Alex would be a better reviewer for this one.

Comment (viktorklang) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:50]: No, I'm completly fine with this PullRequest.
The important part is that the functionality goes into M4, the implementation can be tweaked after that.

**Comment by phaller** [comment]
Alright then. Josh, this is good to pull in then.

**Comment by heathermiller** [comment]
Oops, missed that I was marked as a reviewer on this. LGTM.

**Pull request merged**